sdcm/sct_events/base.py

```diff
@@ -190,6 +191,42 @@ def timestamp(self) -> float:
         """
         return self.source_timestamp or self.event_timestamp
 
+    @cached_property
+    def subcontext_fmt(self):
+        return {}
+
+    def concatenate_subcontext_for_message(self, context_name):
+        s = []
+        for event_context in self.subcontext:
+            if isinstance(event_context, dict) and context_name in event_context:
+                s.append(event_context[context_name])
+            elif isinstance(event_context, SctEvent) and context_name in event_context.subcontext_fmt:
+                s.append(event_context.subcontext_fmt[context_name])
+        return ','.join(s)
+
+    @property
+    def msgfmt(self):
+        fmt = "({0.base} {0.severity}) period_type={0.period_type} event_id={0.event_id}"
+
+        if during_nemesis := self.concatenate_subcontext_for_message(context_name='nemesis_name'):
+            fmt += f" during_nemesis={during_nemesis}"
+
+        return fmt
+
+    def add_subcontext(self, registry):
```

Comment:
if on all calls to this function we are using `ContinuousEventsRegistry()` which is a singleton, why we need this as a parameter here ? do we assume that there would be multiple of those registries ?

and even if so, I would wait until we have that use case, and see how it's fit, and only then extend those functions for it.

Reply:
changed

Reply:
changed